### PR TITLE
pmt: serialization change from htobeXX() to boost::endian

### PR DESCRIPTION
**Pull request by mrjacobagilbert**
this preserves the behavior in a more portable way since endian.h has a variety of portability issues

Addresses #3366 and #3401 and is related to #3320, the backport has been edited to reflect boost::endian #3342

### CODE REVIEW

File: gnuradio-runtime/include/pmt/pmt.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <pmt/api.h>
 #include <stdint.h>
 #include <boost/any.hpp>
+#include <boost/endian/conversion.hpp>
```
Comment (mbr0wn) [inline gnuradio-runtime/include/pmt/pmt.h:17]: Why is this include here?

Comment (jacobgilbert-snl) [reply gnuradio-runtime/include/pmt/pmt.h]: moved to pmt_serialization.cc

**Comment by michaelld** [comment]
@mormj this is a critical bug fix for Windows and OSX. If this LGTY, then go ahead and merge it ASAP ... thx!

**Comment by jwmelto** [comment]
boost/endian/conversion.hpp was introduced in Boost 1.58. gnuradio still allows Boost 1.53 as an acceptable version. This doesn't compile with the older Boost.

**Pull request merged**